File: backend/src/main/java/com/furios/labirinto_dungeonico/api/dto/MoveResponse.java
package com.furios.labirinto_dungeonico.api.dto;

import com.furios.labirinto_dungeonico.combat.CombatEvent;
import com.furios.labirinto_dungeonico.session.GameSession;

import java.util.List;

/**
 * Resposta de {@code POST /api/session/{id}/move}. {@code combatEvents} é {@code null} quando o
 * movimento não desencadeou combate; quando presente, RF-20 a RF-27 já foram aplicados (XP/loot
 * na vitória, RN-01/RN-02 na derrota) antes desta resposta ser montada.
 */
public record MoveResponse(GameSession session, List<CombatEvent> combatEvents) {
}

File: app/lib/core/models/move_response.dart
import 'combat_event.dart';
import 'game_session.dart';

/// Resposta de `POST /api/session/{id}/move`. [combatEvents] é `null` quando
/// o movimento não desencadeou combate.
class MoveResponse {
  const MoveResponse({required this.session, required this.combatEvents});

  factory MoveResponse.fromJson(Map<String, dynamic> json) => MoveResponse(
        session: GameSession.fromJson(json['session'] as Map<String, dynamic>),
        combatEvents: json['combatEvents'] == null
            ? null
            : (json['combatEvents'] as List)
                .map((e) => CombatEvent.fromJson(e as Map<String, dynamic>))
                .toList(),
      );

  final GameSession session;
  final List<CombatEvent>? combatEvents;
}
